libraries/networking/src/NodeList.cpp

```diff
@@ -377,7 +377,7 @@ void NodeList::sendDomainServerCheckIn() {
             // if so we need to make sure we have an up-to-date local port in case it restarted
 
             if (domainSockAddr.getAddress() == QHostAddress::LocalHost
-                || hostname == "localhost") {
+                || hostname == "localhost" && _domainPortAutoDiscovery == true) {
```

Comment:
Mixing && and || here is confusing and wrong.  Please add parenthesis here, specifically as (a || b) && c